### PR TITLE
Fix canonicalization of layer IDs

### PR DESCRIPTION
**Pull request by nalind**
Fix canonicalization of layer IDs when computing layer differences, which we also broke in #6.

### CODE REVIEW

File: storage/layers.go [hunk 1]
```diff
@@ -624,14 +624,21 @@ func (r *layerStore) Changes(from, to string) ([]archive.Change, error) {
 	if !ok {
 		return nil, ErrLayerUnknown
 	}
-	fromLayer, ok := r.lookup(from)
-	if !ok {
-		fromLayer, ok = r.lookup(toLayer.Parent)
+	to = toLayer.ID
```
Comment (rhatdan) [inline storage/layers.go]: It looks like this same block of code is copied three times, shouldn't we just add a new function to handle it?

Comment (nalind) [inline storage/layers.go]: Yeah, that's a better idea.  Fixing that.

**Comment by nalind** [comment]
Okay, this gets buildah's self-tests working again.

**Comment by rhatdan** [comment]
LGTM

Much better

**Comment by nalind** [comment]
Okay, merging it.

**Pull request merged**